dask/array/tests/test_creation.py

```diff
@@ -61,6 +61,41 @@ def test_arr_like(funcname, shape, cast_shape, dtype, cast_chunks, chunks):
         assert (np_r == np.asarray(da_r)).all()
 
 
+@pytest.mark.parametrize(
+    "funcname", ["empty_like", "ones_like", "zeros_like", "full_like",],
+)
+@pytest.mark.parametrize(
+    "shape, chunks, out_shape",
+    [
+        ((10, 10), (4, 4), None),
+        ((10, 10), (4, 4), (20, 3)),
+        ((10, 10), (4), (20)),
+        ((10, 10, 10), (4, 2), (5, 5)),
+        ((2, 3, 5, 7), None, (3, 5, 7)),
+        ((2, 3, 5, 7), (2, 5, 3), (3, 5, 7)),
+        ((2, 3, 5, 7), (2, 5, 3, "auto", 3), (11,) + (2, 3, 5, 7)),
+        ((2, 3, 5, 7), "auto", (3, 5, 7)),
+    ],
+)
+@pytest.mark.parametrize("dtype", ["i4"])
+def test_arr_like_shape(funcname, shape, dtype, chunks, out_shape):
+    np_func = getattr(np, funcname)
+    da_func = getattr(da, funcname)
+    if "full" in funcname:
+        old_np_func = np_func
+        old_da_func = da_func
+
+        np_func = lambda *a, **k: old_np_func(*a, fill_value=5, **k)
+        da_func = lambda *a, **k: old_da_func(*a, fill_value=5, **k)
```

Comment:
Alternatively, you could add a kwargs option to the funcname parametrization

```python
@pytest.mark.parametrize("funcname,kwargs", [("ones_like", {}), ...])
```

Reply:
👍